lib/api/unittest/CBoostedTreeRegressionInferenceModelBuilderTest.cc

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include "CBoostedTreeRegressionInferenceModelBuilderTest.h"
+
+#include <core/CDataAdder.h>
+#include <core/CDataFrame.h>
+#include <core/CDataSearcher.h>
+#include <core/CFloatStorage.h>
+
+#include <maths/CLinearAlgebraEigen.h>
+
+#include <api/CDataFrameAnalysisSpecification.h>
+#include <api/CDataFrameAnalysisSpecificationJsonWriter.h>
+#include <api/CDataFrameAnalyzer.h>
+#include <api/CInferenceModelDefinition.h>
+
+#include <test/CDataFrameTestUtils.h>
+#include <test/CRandomNumbers.h>
+#include <test/CTestTmpDir.h>
+
+#include <fstream>
+#include <string>
+
+using namespace ml;
+
+namespace {
+
+using TStrVec = std::vector<std::string>;
+using TDataAdderUPtr = std::unique_ptr<ml::core::CDataAdder>;
+using TPersisterSupplier = std::function<TDataAdderUPtr()>;
+using TDataSearcherUPtr = std::unique_ptr<ml::core::CDataSearcher>;
+using TRestoreSearcherSupplier = std::function<TDataSearcherUPtr()>;
+using TDataFrameUPtr = std::unique_ptr<ml::core::CDataFrame>;
+using TDoubleVec = std::vector<double>;
+using TDoubleVecVec = std::vector<TDoubleVec>;
+using TPoint = maths::CDenseVector<maths::CFloatStorage>;
+using TPointVec = std::vector<TPoint>;
+using TRowItr = core::CDataFrame::TRowItr;
+using TStrSizeUMap = std::unordered_map<std::string, std::size_t>;
+using TStrSizeUMapVec = std::vector<TStrSizeUMap>;
+
+auto regressionSpec(std::string dependentVariable,
```

Comment:
It is a shame this has got duplicated. Can we leave a TODO to factor this out. I've made some changes in a separate PR which will collide with this, so perhaps better to wait for those are merged before doing this.

Reply:
TODO added.